src/convenience.jl

```diff
@@ -0,0 +1,389 @@
+stored in `x`. If passed `mumps`, must have previously been provided
+with matrix `A`.
+
+See also: [`mumps_select_inv`](@ref), [`get_rhs!`](@ref), [`get_rhs`](@ref)
+"""
+function mumps_select_inv!(x::AbstractSparseArray,mumps::Mumps)
+    set_icntl!(mumps,30,1)
+    set_icntl!(mumps,20,3)
+    provide_rhs!(mumps,x)
+    if mumps.job ∈ [2,4] # if already factored, just solve
+        mumps.job = 3
+    elseif mumps.job ∈ [1] # if analyzed only, factorize and solve
+        mumps.job=5
+    else # else analyze, factor, solve
+        mumps.job=6
+    end
+    invoke_mumps!(mumps)
+    get_rhs!(x,mumps)
+end
+function mumps_select_inv!(x,A)
+    size(x)==size(A) ? nothing : throw(MUMPSException("target and matrix must have same size"))
+    mumps = Mumps(A)
+    suppress_display!(mumps)
+    mumps_select_inv!(x,mumps)
+    finalize!(mumps)
+end
+"""
+    mumps_select_inv(A,x) -> A⁻¹
+    mumps_select_inv(A,I,J) -> A⁻¹
+
+Compute selected elements of A⁻¹.
+If two arguments are passed, the second must be sparse, and its sparsity pattern
+determines the entries of the inverse.
+If three arguments are passed, the integer arrays `I` and `J` specify which
+entries via `i(k),j(k) = I[k],J[k]`.
+
+See also: [`mumps_select_inv!`](@ref)
+"""
+function mumps_select_inv(A,x::AbstractSparseArray)
+    y = copy(x)
+    mumps_select_inv!(y,A)
+    return y
+end
+function mumps_select_inv(A,I::AbstractArray{Int},J::AbstractArray{Int})
+    x = sparse(I,J,fill(one(eltype(A)),length(I)),size(A)...)
+    mumps_select_inv!(x,A)
+    return x
+end
+
+
+"""
+    initialize!(mumps)
+
+Reinitialize `mumps`, regardless of its current state
+
+See also: [`finalize`](@ref)
+"""
+function initialize!(mumps::Mumps)
+    suppress_display!(mumps)
+    mumps._finalized=false
+    mumps.job=-1
+    invoke_mumps!(mumps)
+end
+
+
+"""
+    finalize!(mumps)
+
+Release the pointers contained in `mumps`
+
+See also: [`initialize`](@ref)
+"""
+function finalize!(mumps::Mumps)
+    suppress_display!(mumps)
+    return is_finalized(mumps) ? nothing : finalize_unsafe!(mumps)
+end
+function finalize_unsafe!(mumps::Mumps)
+    mumps._finalized=true
+    mumps.job=-2
+    invoke_mumps_unsafe!(mumps)
+    return nothing
+end
+
+
+function Base.:\(mumps::Mumps,y)
+    suppress_display!(mumps)
+    if mumps.job < 2
+        mumps_factorize!(mumps)
+    end
+    x = mumps_solve(mumps,y)
+    return x
+end
+function LinearAlgebra.ldiv!(mumps::Mumps,y)
+    suppress_display!(mumps)
+    if mumps.job < 2
+        mumps_factorize!(mumps)
+    else
+        mumps.job=2
+    end
+    provide_rhs!(mumps,y)
+    mumps_solve!(y,mumps)
+end
+function LinearAlgebra.ldiv!(x,mumps::Mumps,y)
+    suppress_display!(mumps)
+    if mumps.job < 2
+        mumps_factorize!(mumps)
+    end
+    mumps_solve!(x,mumps,y)
+end
+function LinearAlgebra.inv(mumps::Mumps)
+    suppress_display!(mumps)
+    y = sparse(1:mumps.n,1:mumps.n,1:mumps.n,mumps.n,mumps.n)
```

Comment:
This creates a diagonal matrix, but `inv` is supposed to return the entire inverse, isn't it?